pandas/_libs/tslibs/timezones.pyx

```diff
@@ -102,6 +102,14 @@ cpdef inline tzinfo maybe_get_tz(object tz):
             # On Python 3 on Windows, the filename is not always set correctly.
             if isinstance(tz, _dateutil_tzfile) and '.tar.gz' in tz._filename:
                 tz._filename = zone
+        elif tz[0] in {'-', '+'}:
```

Comment:
what does the current path e.g. `pytz.timezone(tz)` do with one of these (vs what you are showing). e.g. why fundamentally is this a bug

Reply:
It can't read timezones without names. So if timezone is `Europe/Lisbon` it works, but if it's `+01` it raises exceptions upon reading a parquet file with timezone aware date index. See the tests, all of them throws exceptions without the new patch.

Reply:
@alippai jeff is asking for roughly the same thing i asked before: what _pandas_ function behaves incorrectly?

Reply:
Sorry, `pd.read_parquet` serialized by pandas (using both pyarrow and fastparquet)

Reply:
I'm confused about this a little bit, the title and description of this issue links to a minimal example to reproduce the bug. GitHub links the issue in the right sidebar too. Should I organize the issue + PR differently next time I find a bug?

The two examples:
```python
idx = [datetime.strptime('2019-01-04T16:41:24+0200', "%Y-%m-%dT%H:%M:%S%z")]
df = pd.DataFrame(index=idx)
df.to_parquet("out.parquet", engine="fastparquet")  # <-- Exception, can't convert & write timezone `UTC+02:00`
```
```python
idx = [datetime.now(timezone.utc)]
df = pd.DataFrame(index=idx)
df.to_parquet('out.parquet')
pd.read_parquet('out.parquet')  # <-- Exception, timezone is written as `+00:00`, Pandas can't read it
```

Reply:
```
import pytz
print(pytz.timezone('+00:00'))  # pytz.exceptions.UnknownTimeZoneError: '+00:00'
print(pytz.timezone('UTC+02:00'))  # pytz.exceptions.UnknownTimeZoneError: 'UTC+02:00'
```

Reply:
> I'm confused about this a little bit, the title and description of this issue links to a minimal example to reproduce the bug.

Poor wording on my part.  I'm asking for a way to demonstrate the incorrect behavior that doesn't depend on pyarrow/fastparquet.

It would be helpful to localize the bug to something "closer" to the code being changed.  `pd.read_parquet` is not directly calling `maybe_get_tz`, so it isn't obvious how the bug is connected to the code being changed (or if the code being changed is the only/best way to fix the bug)

Reply:
This makes it clearer, thanks. So both `tz_localize()` and `tz_convert()` used by fastparquet and pyarrow helpers is fed with non-tz parameters and generally this is accepted:
```python
pd.DatetimeIndex([datetime(2020, 9, 5)]).tz_localize('Europe/Lisbon') # works
pd.DatetimeIndex([datetime(2020, 9, 5)]).tz_localize('+01:00') # doesn't work
pd.DatetimeIndex([datetime(2020, 9, 5)]).tz_localize('UTC+01:00') # doesn't work
```
So my patch fixes this asymmetry. Like you don't have to write
```python 
pd.DatetimeIndex([datetime(2020, 9, 5)]).tz_localize(pytz.timezone('Europe/Lisbon'))
```
Both fastparquet and pyarrow feeds the timezone directly into `tz_localize` and `tz_convert`. When they do it with values like `US/Eastern` this works, then they have values like `+01:00` or `UTC+01:00` it raises exceptions. Both `tz_localize` and `tz_convert` depend on `maybe_get_tz` directly (where I added my change).

Reply:
@jbrockmendel does this add more depth and context to the issue?

Reply:
That is helpful, thanks.  So for the PR, can you take those currently not-working pd.DatetimeIndex lines and make tests for them?  (i guess for Timestamp too)

Reply:
That sounds useful, I'll add the tests tomorrow. For the pyarrow and fastparquet I'll keep the new "high-level" tests too to ensure they don't break this use-case as timezone handling is a custom/fragile part.

Reply:
Added the new timezeones to `TIMEZONES = []` in conftest.py, that hits the `DatetimeIndex` constructor and other related code paths.